pkg/streamingpromql/operators/aggregations/aggregation.go

```diff
@@ -263,7 +277,17 @@ func (a *Aggregation) emitAnnotation(generator types.AnnotationGenerator) {
 	a.Annotations.Add(generator(metricName, a.Inner.ExpressionPosition()))
 }
 
+func (a *Aggregation) emitAnnotationParam(param float64, generator types.AnnotationParamGenerator) {
+	a.Annotations.Add(generator(param, a.Param.ExpressionPosition()))
+}
+
 func (a *Aggregation) Close() {
+	if len(a.paramData.Samples) > 0 {
```

Comment:
[nit] This can be simplified to:

```suggestion
	if a.paramData.Samples != nil {
```